docs/guide/controller-programming.md

```diff
@@ -452,6 +452,53 @@ if (finished) {
 
 Note that the exit status as well as the value returned by the `main` function are ignored by Webots.
 
+### Console Output
+
+As mentioned earlier, printing to `stdout` or `stderr` from a controller will be redirected by default to the Webots console.
+Note however, that the Webots console doesn't support `stdin` input.
+Like most terminals, it supports a few basic [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) for setting text styles and clearing the content of the console:
+  - 3-bit color (foreground and background)
+  - Bold style
+  - Underline style
+  - Clear screen (same as issuing `clear` command in your terminal)
+  - Reset (colors and styles)
+
+To demonstrate how to use those, there is an example world and a controller file respectively located in "[WEBOTS\_HOME/projects/samples/howto/worlds/console.wbt](https://github.com/cyberbotics/webots/tree/master/projects/samples/howto/worlds/console.wbt)" and "[WEBOTS\_HOME/projects/samples/howto/controllers/console/console.c](https://github.com/cyberbotics/webots/tree/master/projects/samples/howto/controllers/console/console.c)".
+
+The related C header is located at "[WEBOTS\_HOME/include/controller/c/webots/utils/ansi\_codes.h](https://github.com/cyberbotics/webots/tree/master/include/controller/c/webots/utils/ansi_codes.h)", it contains some useful macros on top of constants, to use it:
+```c
+#include <webots/utils/ansi_codes.h>
+```
+
+A C++ header is also available as a wrapper of the above one. Use the provided constants with `std::cout` at your wish.
+```cpp
+#include <AnsiCodes.hpp>
+```
+
+For Python, `import controller` and access the symbols like this: `controller.ANSI_GREEN_FOREGROUND`.
+
+Tip: You can `import controller as ctl` and shorten it to `ctl.ANSI_GREEN_FOREGROUND`.
+```python
+import controller as ctl
+
+print(ctl.ANSI_GREEN_FOREGROUND + "Hello World!")
+```
+
+For Java, you may want to create your own style in a constant string for example:
+```java
+import com.cyberbotics.webots.controller.wrapperConstants;
```

Comment:
This is not nice to refer to global wrapper constants...
It's not meaningful with respect to ANSI codes.
I would prefer to see something like: `import com.cyberbotics.webots.controller.AnsiCodes;`
And then use `final String format = AnsiCodes.GREEN_FOREGROUND;`.

Reply:
Yes, however how to rename the wrapperConstants package to AnsiCodes regarding the [controller.i](https://github.com/cyberbotics/webots/blob/master/resources/languages/java/controller.i), should I create a new module specifically next to the already existing `%module wrapper` ?

Reply:
I don't know yet how to achieve this. I am investigating...